tests/providers/microsoft/azure/hooks/test_azure_data_factory.py

```diff
@@ -177,7 +177,7 @@ def test_create_factory(hook: AzureDataFactoryHook, user_args, sdk_args):
     implicit_factory=((MODEL,), (DEFAULT_RESOURCE_GROUP, DEFAULT_FACTORY, MODEL)),
 )
 def test_update_factory(hook: AzureDataFactoryHook, user_args, sdk_args):
-    hook._factory_exists = Mock(return_value=True)
+    hook._factory_exists = Mock(return_value=True)  # type: ignore
```

Comment:
I think we should find a much better way to have mypy satisfied here.

I think we shoudl (if that works)  add spec='boolean' to mock's constructor?

Reply:
A `patch.object(...)` context manager does the trick. WDYT?